### PR TITLE
HDDS-4232. Use single thread for KeyDeletingService.

### PR DESCRIPTION
**Pull request by lokeshj1703**
## What changes were proposed in this pull request?

KeyDeletingService scan the keys from a particular rocksdb table and sends deletion request to SCM. Every thread would scan the table and send deletion requests. This can lead to multiple deletion request for a particular block. There is currently no way to distribute the keys to be deleted amongst multiple threads.

## What is the link to the Apache JIRA

HDDS-4232

## How was this patch tested?

This lead to failure in TestBlockDeletion. Please refer https://github.com/apache/hadoop-ozone/pull/1121/commits/49b610a9feb54899ae226d3b6f2e36c5f8cc784f

### CODE REVIEW

File: hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/KeyDeletingService.java [hunk 1]
```diff
@@ -66,9 +66,6 @@
   private static final Logger LOG =
       LoggerFactory.getLogger(KeyDeletingService.class);
 
-  // The thread pool size for key deleting service.
-  private final static int KEY_DELETING_CORE_POOL_SIZE = 2;
```
Comment (amaliujia) [inline hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/KeyDeletingService.java:70]: when just not change 2 to 1?

**Comment by bharatviswa504** [comment]
Hi @lokeshj1703 
https://issues.apache.org/jira/browse/HDDS-4121 This Jira changes the behavior with this openKeyCleanup service does not call SCM directly, it moves the keys to deleteTable, and it will be picked by KeyDeletingService and it will be only the one to send delete keys request to SCM from OM.

**Comment by bharatviswa504** [comment]
And also a question, these are expired Keys in the openKeyTable, how for the same block 2 threads can send the request before. And also I see there is no logic for deleteExpiredKey today, and openKeyCleanup is not yet integrated in OM.

**Comment by lokeshj1703** [comment]
The PR requires change in 1414 to address the test failure.

File: hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/KeyDeletingService.java (hunk 1 above)
Comment (lokeshj1703) [reply hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/KeyDeletingService.java:70]: Addressed it in latest commit.

**Comment by lokeshj1703** [comment]
> [HDDS-4120](https://issues.apache.org/jira/browse/HDDS-4120) This Jira changes the behavior of openKeyCleanup service, and does not call SCM directly, it moves the keys to deleteTable by making a write request to OM. And later it will be picked by KeyDeletingService and it will be only the one to send delete keys request to SCM from OM.

@bharatviswa504 Thanks for providing the info! In that case we do not need the changes for openKeyCleanupService. I have addressed it in latest commit.

File: hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/KeyDeletingService.java (hunk 1 above)
Comment (amaliujia) [reply hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/KeyDeletingService.java:70]: Thanks!

**Review by adoroszlai — APPROVED** [review]

**Comment by adoroszlai** [comment]
Thanks @lokeshj1703 for the fix and @amaliujia and @bharatviswa504 for the review.

**Pull request merged**